### ISSUE TITLE
Stuck at "loading packages"

### ISSUE THREAD

**Issue by MrBananaPants** [buresdv/Cork#111]
I just compiled Cork from source.
Everything works fine except the home screen which is stuck at "loading packages".

Maybe the issue could be related to the macOS 14 Developer beta I'm using?
Mac Mini M2, macOS 14 Dev beta 1.

![Screenshot 2023-06-11 at 15 33 35](https://github.com/buresdv/Cork/assets/41532982/c7d72568-2f73-4135-9032-06b19c6085c8)

**Comment by buresdv** [comment]
It's possible that this might be a beta thing. "Loading Packages" is shown until packages and casks are loaded, but I see that you have them loaded in the sidebar, so it can't be that. The only other condition is that you have no taps added, but that's not possible because you have packages installed at all.

Could you please run Cork from Xcode and send me the console output? I'll have a look what's up 😊

**Comment by MrBananaPants** [comment]
@buresdv thank you for the quick response!
Here's the console output:

```
Brew executable path: /opt/homebrew/bin/brew
Documents directory: /Users/joranvancoillie/Documents/Cork
Documents directory exists
Metadata file exists
Started Cellar task at 2023-06-11 13:54:54 +0000
Scanned version: file:///opt/homebrew/Cellar/gettext/0.21.1/
Found desirable version: file:///opt/homebrew/Cellar/gettext/0.21.1/. Appending to temporary package list
URL to package gettext is [/opt/homebrew/Cellar/gettext/0.21.1/]
URL of this package: /opt/homebrew/Cellar/gettext/

 Installation date for package gettext at path /opt/homebrew/Cellar/gettext/ is 2023-04-27 20:47:41 +0000 

Package gettext was not installed intentionally
Scanned version: file:///opt/homebrew/Cellar/pcre2/10.42/
Found desirable version: file:///opt/homebrew/Cellar/pcre2/10.42/. Appending to temporary package list
URL to package pcre2 is [/opt/homebrew/Cellar/pcre2/10.42/]
URL of this package: /opt/homebrew/Cellar/pcre2/

 Installation date for package pcre2 at path /opt/homebrew/Cellar/pcre2/ is 2023-04-27 20:47:43 +0000 

Package pcre2 was not installed intentionally
Scanned version: file:///opt/homebrew/Cellar/git/2.41.0/
Found desirable version: file:///opt/homebrew/Cellar/git/2.41.0/. Appending to temporary package list
URL to package git is [/opt/homebrew/Cellar/git/2.41.0/]
URL of this package: /opt/homebrew/Cellar/git/

 Installation date for package git at path /opt/homebrew/Cellar/git/ is 2023-04-27 20:47:44 +0000 

Package git was installed intentionally
Failed while getting package version: Error Domain=NSCocoaErrorDomain Code=256 "The file “.keepme” couldn’t be opened." UserInfo={NSURL=/opt/homebrew/Cellar/.keepme/, NSFilePath=/opt/homebrew/Cellar/.keepme, NSUnderlyingError=0x600002d6a250 {Error Domain=NSPOSIXErrorDomain Code=20 "Not a directory"}}
Started Cask task at 2023-06-11 13:54:54 +0000
Scanned version: file:///opt/homebrew/Caskroom/microsoft-teams/1.6.00.12966/
Found desirable version: file:///opt/homebrew/Caskroom/microsoft-teams/1.6.00.12966/. Appending to temporary package list
URL to package microsoft-teams is [/opt/homebrew/Caskroom/microsoft-teams/1.6.00.12966/]
URL of this package: /opt/homebrew/Caskroom/microsoft-teams/

 Installation date for package microsoft-teams at path /opt/homebrew/Caskroom/microsoft-teams/ is 2023-04-27 20:35:50 +0000 

Scanned version: file:///opt/homebrew/Caskroom/visual-studio-code/1.79.0/
Found desirable version: file:///opt/homebrew/Caskroom/visual-studio-code/1.79.0/. Appending to temporary package list
URL to package visual-studio-code is [/opt/homebrew/Caskroom/visual-studio-code/1.79.0/]
URL of this package: /opt/homebrew/Caskroom/visual-studio-code/

 Installation date for package visual-studio-code at path /opt/homebrew/Caskroom/visual-studio-code/ is 2023-02-23 13:21:43 +0000 

Scanned version: file:///opt/homebrew/Caskroom/google-chrome/114.0.5735.106/
Found desirable version: file:///opt/homebrew/Caskroom/google-chrome/114.0.5735.106/. Appending to temporary package list
URL to package google-chrome is [/opt/homebrew/Caskroom/google-chrome/114.0.5735.106/]
URL of this package: /opt/homebrew/Caskroom/google-chrome/

 Installation date for package google-chrome at path /opt/homebrew/Caskroom/google-chrome/ is 2023-06-08 08:51:18 +0000 

Scanned version: file:///opt/homebrew/Caskroom/discord/0.0.275/
Found desirable version: file:///opt/homebrew/Caskroom/discord/0.0.275/. Appending to temporary package list
URL to package discord is [/opt/homebrew/Caskroom/discord/0.0.275/]
URL of this package: /opt/homebrew/Caskroom/discord/

 Installation date for package discord at path /opt/homebrew/Caskroom/discord/ is 2023-04-27 20:24:34 +0000 

Scanned version: file:///opt/homebrew/Caskroom/adobe-creative-cloud/5.10.0.573/
Found desirable version: file:///opt/homebrew/Caskroom/adobe-creative-cloud/5.10.0.573/. Appending to temporary package list
URL to package adobe-creative-cloud is [/opt/homebrew/Caskroom/adobe-creative-cloud/5.10.0.573/]
URL of this package: /opt/homebrew/Caskroom/adobe-creative-cloud/

 Installation date for package adobe-creative-cloud at path /opt/homebrew/Caskroom/adobe-creative-cloud/ is 2023-06-02 08:26:00 +0000 

Scanned version: file:///opt/homebrew/Caskroom/send-anywhere/23.5.18/
Found desirable version: file:///opt/homebrew/Caskroom/send-anywhere/23.5.18/. Appending to temporary package list
URL to package send-anywhere is [/opt/homebrew/Caskroom/send-anywhere/23.5.18/]
URL of this package: /opt/homebrew/Caskroom/send-anywhere/

 Installation date for package send-anywhere at path /opt/homebrew/Caskroom/send-anywhere/ is 2023-02-23 13:25:23 +0000 

Scanned version: file:///opt/homebrew/Caskroom/android-platform-tools/34.0.3/
Found desirable version: file:///opt/homebrew/Caskroom/android-platform-tools/34.0.3/. Appending to temporary package list
URL to package android-platform-tools is [/opt/homebrew/Caskroom/android-platform-tools/34.0.3/]
URL of this package: /opt/homebrew/Caskroom/android-platform-tools/

 Installation date for package android-platform-tools at path /opt/homebrew/Caskroom/android-platform-tools/ is 2023-02-23 13:06:21 +0000 

Scanned version: file:///opt/homebrew/Caskroom/visual-paradigm/17.1,20230601/
Found desirable version: file:///opt/homebrew/Caskroom/visual-paradigm/17.1,20230601/. Appending to temporary package list
URL to package visual-paradigm is [/opt/homebrew/Caskroom/visual-paradigm/17.1,20230601/]
URL of this package: /opt/homebrew/Caskroom/visual-paradigm/

 Installation date for package visual-paradigm at path /opt/homebrew/Caskroom/visual-paradigm/ is 2023-02-23 13:07:44 +0000 

Scanned version: file:///opt/homebrew/Caskroom/appcleaner/3.6.7/
Found desirable version: file:///opt/homebrew/Caskroom/appcleaner/3.6.7/. Appending to temporary package list
URL to package appcleaner is [/opt/homebrew/Caskroom/appcleaner/3.6.7/]
URL of this package: /opt/homebrew/Caskroom/appcleaner/

 Installation date for package appcleaner at path /opt/homebrew/Caskroom/appcleaner/ is 2023-02-23 13:23:16 +0000 

Scanned version: file:///opt/homebrew/Caskroom/onedrive/23.101.0514.0001/
Found desirable version: file:///opt/homebrew/Caskroom/onedrive/23.101.0514.0001/. Appending to temporary package list
URL to package onedrive is [/opt/homebrew/Caskroom/onedrive/23.101.0514.0001/]
URL of this package: /opt/homebrew/Caskroom/onedrive/

 Installation date for package onedrive at path /opt/homebrew/Caskroom/onedrive/ is 2023-04-27 20:39:41 +0000 

Scanned version: file:///opt/homebrew/Caskroom/spotify/1.2.13.661,a588f749,984/
Found desirable version: file:///opt/homebrew/Caskroom/spotify/1.2.13.661,a588f749,984/. Appending to temporary package list
URL to package spotify is [/opt/homebrew/Caskroom/spotify/1.2.13.661,a588f749,984/]
URL of this package: /opt/homebrew/Caskroom/spotify/

 Installation date for package spotify at path /opt/homebrew/Caskroom/spotify/ is 2023-06-09 18:05:50 +0000 

Scanned version: file:///opt/homebrew/Caskroom/hex-fiend/2.16.0/
Found desirable version: file:///opt/homebrew/Caskroom/hex-fiend/2.16.0/. Appending to temporary package list
URL to package hex-fiend is [/opt/homebrew/Caskroom/hex-fiend/2.16.0/]
URL of this package: /opt/homebrew/Caskroom/hex-fiend/

 Installation date for package hex-fiend at path /opt/homebrew/Caskroom/hex-fiend/ is 2023-02-23 13:12:54 +0000 

Scanned version: file:///opt/homebrew/Caskroom/mos/3.4.1/
Found desirable version: file:///opt/homebrew/Caskroom/mos/3.4.1/. Appending to temporary package list
URL to package mos is [/opt/homebrew/Caskroom/mos/3.4.1/]
URL of this package: /opt/homebrew/Caskroom/mos/

 Installation date for package mos at path /opt/homebrew/Caskroom/mos/ is 2023-02-23 13:25:31 +0000 

Scanned version: file:///opt/homebrew/Caskroom/anki/2.1.65/
Found desirable version: file:///opt/homebrew/Caskroom/anki/2.1.65/. Appending to temporary package list
URL to package anki is [/opt/homebrew/Caskroom/anki/2.1.65/]
URL of this package: /opt/homebrew/Caskroom/anki/

 Installation date for package anki at path /opt/homebrew/Caskroom/anki/ is 2023-02-23 13:22:43 +0000 

Scanned version: file:///opt/homebrew/Caskroom/macs-fan-control/1.5.15/
Found desirable version: file:///opt/homebrew/Caskroom/macs-fan-control/1.5.15/. Appending to temporary package list
URL to package macs-fan-control is [/opt/homebrew/Caskroom/macs-fan-control/1.5.15/]
URL of this package: /opt/homebrew/Caskroom/macs-fan-control/

 Installation date for package macs-fan-control at path /opt/homebrew/Caskroom/macs-fan-control/ is 2023-02-23 13:25:33 +0000 

Scanned version: file:///opt/homebrew/Caskroom/eclipse-ide/4.27.0,2023-03/
Found desirable version: file:///opt/homebrew/Caskroom/eclipse-ide/4.27.0,2023-03/. Appending to temporary package list
URL to package eclipse-ide is [/opt/homebrew/Caskroom/eclipse-ide/4.27.0,2023-03/]
URL of this package: /opt/homebrew/Caskroom/eclipse-ide/

 Installation date for package eclipse-ide at path /opt/homebrew/Caskroom/eclipse-ide/ is 2023-02-23 13:06:53 +0000 

Scanned version: file:///opt/homebrew/Caskroom/notion/2.1.15/
Found desirable version: file:///opt/homebrew/Caskroom/notion/2.1.15/. Appending to temporary package list
URL to package notion is [/opt/homebrew/Caskroom/notion/2.1.15/]
URL of this package: /opt/homebrew/Caskroom/notion/

 Installation date for package notion at path /opt/homebrew/Caskroom/notion/ is 2023-02-23 13:22:56 +0000 

Scanned version: file:///opt/homebrew/Caskroom/rectangle/0.69/
Found desirable version: file:///opt/homebrew/Caskroom/rectangle/0.69/. Appending to temporary package list
URL to package rectangle is [/opt/homebrew/Caskroom/rectangle/0.69/]
URL of this package: /opt/homebrew/Caskroom/rectangle/

 Installation date for package rectangle at path /opt/homebrew/Caskroom/rectangle/ is 2023-02-23 13:25:18 +0000 

Scanned version: file:///opt/homebrew/Caskroom/garmin-express/7.17.1.0,7170100/
Found desirable version: file:///opt/homebrew/Caskroom/garmin-express/7.17.1.0,7170100/. Appending to temporary package list
URL to package garmin-express is [/opt/homebrew/Caskroom/garmin-express/7.17.1.0,7170100/]
URL of this package: /opt/homebrew/Caskroom/garmin-express/

 Installation date for package garmin-express at path /opt/homebrew/Caskroom/garmin-express/ is 2023-04-27 20:19:21 +0000 

Scanned version: file:///opt/homebrew/Caskroom/microsoft-word/16.73.23051401/
Found desirable version: file:///opt/homebrew/Caskroom/microsoft-word/16.73.23051401/. Appending to temporary package list
URL to package microsoft-word is [/opt/homebrew/Caskroom/microsoft-word/16.73.23051401/]
URL of this package: /opt/homebrew/Caskroom/microsoft-word/

 Installation date for package microsoft-word at path /opt/homebrew/Caskroom/microsoft-word/ is 2023-04-27 20:37:44 +0000 

Scanned version: file:///opt/homebrew/Caskroom/onyx/4.4.0/
Found desirable version: file:///opt/homebrew/Caskroom/onyx/4.4.0/. Appending to temporary package list
URL to package onyx is [/opt/homebrew/Caskroom/onyx/4.4.0/]
URL of this package: /opt/homebrew/Caskroom/onyx/

 Installation date for package onyx at path /opt/homebrew/Caskroom/onyx/ is 2023-02-23 13:13:13 +0000 

Scanned version: file:///opt/homebrew/Caskroom/todoist/8.3.2/
Found desirable version: file:///opt/homebrew/Caskroom/todoist/8.3.2/. Appending to temporary package list
URL to package todoist is [/opt/homebrew/Caskroom/todoist/8.3.2/]
Scanned version: file:///opt/homebrew/Caskroom/todoist/8.2.3/
Found desirable version: file:///opt/homebrew/Caskroom/todoist/8.2.3/. Appending to temporary package list
URL to package todoist is [/opt/homebrew/Caskroom/todoist/8.3.2/, /opt/homebrew/Caskroom/todoist/8.2.3/]
URL of this package: /opt/homebrew/Caskroom/todoist/

 Installation date for package todoist at path /opt/homebrew/Caskroom/todoist/ is 2023-02-23 13:23:04 +0000 

Scanned version: file:///opt/homebrew/Caskroom/pycharm-ce/2023.1.2,231.9011.38/
Found desirable version: file:///opt/homebrew/Caskroom/pycharm-ce/2023.1.2,231.9011.38/. Appending to temporary package list
URL to package pycharm-ce is [/opt/homebrew/Caskroom/pycharm-ce/2023.1.2,231.9011.38/]
URL of this package: /opt/homebrew/Caskroom/pycharm-ce/

 Installation date for package pycharm-ce at path /opt/homebrew/Caskroom/pycharm-ce/ is 2023-04-27 20:27:14 +0000 

Scanned version: file:///opt/homebrew/Caskroom/microsoft-auto-update/4.59.23051401/
Found desirable version: file:///opt/homebrew/Caskroom/microsoft-auto-update/4.59.23051401/. Appending to temporary package list
URL to package microsoft-auto-update is [/opt/homebrew/Caskroom/microsoft-auto-update/4.59.23051401/]
URL of this package: /opt/homebrew/Caskroom/microsoft-auto-update/

 Installation date for package microsoft-auto-update at path /opt/homebrew/Caskroom/microsoft-auto-update/ is 2023-04-27 20:37:36 +0000 

Scanned version: file:///opt/homebrew/Caskroom/logi-options-plus/1.42.411539/
Found desirable version: file:///opt/homebrew/Caskroom/logi-options-plus/1.42.411539/. Appending to temporary package list
URL to package logi-options-plus is [/opt/homebrew/Caskroom/logi-options-plus/1.42.411539/]
URL of this package: /opt/homebrew/Caskroom/logi-options-plus/

 Installation date for package logi-options-plus at path /opt/homebrew/Caskroom/logi-options-plus/ is 2023-04-27 20:19:51 +0000 

Scanned version: file:///opt/homebrew/Caskroom/teamviewer/15.42.7/
Found desirable version: file:///opt/homebrew/Caskroom/teamviewer/15.42.7/. Appending to temporary package list
URL to package teamviewer is [/opt/homebrew/Caskroom/teamviewer/15.42.7/]
URL of this package: /opt/homebrew/Caskroom/teamviewer/

 Installation date for package teamviewer at path /opt/homebrew/Caskroom/teamviewer/ is 2023-02-23 13:13:57 +0000 

Scanned version: file:///opt/homebrew/Caskroom/google-drive/76.0.3/
Found desirable version: file:///opt/homebrew/Caskroom/google-drive/76.0.3/. Appending to temporary package list
URL to package google-drive is [/opt/homebrew/Caskroom/google-drive/76.0.3/]
URL of this package: /opt/homebrew/Caskroom/google-drive/

 Installation date for package google-drive at path /opt/homebrew/Caskroom/google-drive/ is 2023-02-23 13:10:04 +0000 

Scanned version: file:///opt/homebrew/Caskroom/coconutbattery/3.9.11,6E294072/
Found desirable version: file:///opt/homebrew/Caskroom/coconutbattery/3.9.11,6E294072/. Appending to temporary package list
URL to package coconutbattery is [/opt/homebrew/Caskroom/coconutbattery/3.9.11,6E294072/]
URL of this package: /opt/homebrew/Caskroom/coconutbattery/

 Installation date for package coconutbattery at path /opt/homebrew/Caskroom/coconutbattery/ is 2023-02-23 13:09:48 +0000 

Scanned version: file:///opt/homebrew/Caskroom/mysqlworkbench/8.0.33/
Found desirable version: file:///opt/homebrew/Caskroom/mysqlworkbench/8.0.33/. Appending to temporary package list
URL to package mysqlworkbench is [/opt/homebrew/Caskroom/mysqlworkbench/8.0.33/]
URL of this package: /opt/homebrew/Caskroom/mysqlworkbench/

 Installation date for package mysqlworkbench at path /opt/homebrew/Caskroom/mysqlworkbench/ is 2023-02-23 13:08:25 +0000 

Scanned version: file:///opt/homebrew/Caskroom/slack/4.32.127/
Found desirable version: file:///opt/homebrew/Caskroom/slack/4.32.127/. Appending to temporary package list
URL to package slack is [/opt/homebrew/Caskroom/slack/4.32.127/]
URL of this package: /opt/homebrew/Caskroom/slack/

 Installation date for package slack at path /opt/homebrew/Caskroom/slack/ is 2023-06-07 17:00:36 +0000 

Scanned version: file:///opt/homebrew/Caskroom/vpn-by-google-one/1.6.0.2/
Found desirable version: file:///opt/homebrew/Caskroom/vpn-by-google-one/1.6.0.2/. Appending to temporary package list
URL to package vpn-by-google-one is [/opt/homebrew/Caskroom/vpn-by-google-one/1.6.0.2/]
URL of this package: /opt/homebrew/Caskroom/vpn-by-google-one/

 Installation date for package vpn-by-google-one at path /opt/homebrew/Caskroom/vpn-by-google-one/ is 2023-04-27 20:41:33 +0000 

Scanned version: file:///opt/homebrew/Caskroom/fleet/1.19.95/
Found desirable version: file:///opt/homebrew/Caskroom/fleet/1.19.95/. Appending to temporary package list
URL to package fleet is [/opt/homebrew/Caskroom/fleet/1.19.95/]
URL of this package: /opt/homebrew/Caskroom/fleet/

 Installation date for package fleet at path /opt/homebrew/Caskroom/fleet/ is 2023-04-27 20:25:06 +0000 

Scanned version: file:///opt/homebrew/Caskroom/alfred/5.1.1,2138/
Found desirable version: file:///opt/homebrew/Caskroom/alfred/5.1.1,2138/. Appending to temporary package list
URL to package alfred is [/opt/homebrew/Caskroom/alfred/5.1.1,2138/]
URL of this package: /opt/homebrew/Caskroom/alfred/

 Installation date for package alfred at path /opt/homebrew/Caskroom/alfred/ is 2023-06-02 08:26:26 +0000 

Scanned version: file:///opt/homebrew/Caskroom/microsoft-excel/16.73.23051401/
Found desirable version: file:///opt/homebrew/Caskroom/microsoft-excel/16.73.23051401/. Appending to temporary package list
URL to package microsoft-excel is [/opt/homebrew/Caskroom/microsoft-excel/16.73.23051401/]
URL of this package: /opt/homebrew/Caskroom/microsoft-excel/

 Installation date for package microsoft-excel at path /opt/homebrew/Caskroom/microsoft-excel/ is 2023-04-27 20:34:13 +0000 

Scanned version: file:///opt/homebrew/Caskroom/utm/4.2.5/
Found desirable version: file:///opt/homebrew/Caskroom/utm/4.2.5/. Appending to temporary package list
URL to package utm is [/opt/homebrew/Caskroom/utm/4.2.5/]
URL of this package: /opt/homebrew/Caskroom/utm/

 Installation date for package utm at path /opt/homebrew/Caskroom/utm/ is 2023-02-26 17:32:29 +0000 

Contents of tap folder: []
Loaded name set: [""]
Tagged packages in appState: [""]
Will attempt to place package name 
 not found in Formulae
 not found in Casks
Analytics are ENABLED
Outdated packages output: 
```

**Comment by buresdv** [comment]
No problem! I have to keep the users happy 😆 

The problem seems to be this line: `Contents of tap folder: []`

It looks like it's not finding any taps in the designated tap folder. Could you check if the folder `/opt/homebrew/Library/Taps` exists for you, and if there's anything in it? And do the same with `/usr/local/Homebrew/Library/Taps`, just to be sure

**Comment by MrBananaPants** [comment]
Hi, the folder `/opt/homebrew/Library/Taps` exists but is empty and the folder `/usr/local/Homebrew/Library/Taps` does not exist.

I can use Homebrew in the terminal without any issues (install casks, remove them, upgrade casks and fomulae,...)

**Comment by buresdv** [comment]
Thank yo ufor the info! That's strange, I've never seen Homebrew work with no taps added. Can you do `brew update && brew upgrade` and see if you can still install stuff?

**Comment by MrBananaPants** [comment]
I already had the latest version installed (`4.0.21`) and I'm able to install and upgrade casks without any issues. `brew doctor` also shows no issues (except for me running a developer beta version of macOS).

**Comment by buresdv** [comment]
That's really strange, can you do `brew tap homebrew/core` and see if that fixes the loading problem?

**Comment by MrBananaPants** [comment]
`brew tap homebrew/core` did fix the loading problem but `brew doctor` now gives me the following message:
```
Warning: You have an unnecessary local Core tap!
This can cause problems installing up-to-date formulae.
Please remove it by running:
 brew untap homebrew/core
```

**Comment by buresdv** [comment]
I have never seen this before. I will try to search up what might be wrong, so I'll get back to you later.

**Comment by MrBananaPants** [comment]
I think this might be useful:
[https://github.com/orgs/Homebrew/discussions/4495](https://github.com/orgs/Homebrew/discussions/4495)

**Comment by buresdv** [comment]
Hmm, in that case, do what `brew doctor` tells you to do so you don't end up with a broken Homebrew, and I'll get back to you. If you figure out anything else, please add another comment!

**Comment by buresdv** [comment]
Hmm, in that case, do what `brew doctor` tells you to do so you don't end up with a broken Homebrew, and I'll get back to you. If you figure out anything else, please add another comment!

**Comment by hecaex** [comment]
This happened to me too when I removed the only extra tap that I had previously added. Adding `homebrew/core` and running `brew doctor` results in a warning of multiple `homebre/core` taps. Readding a tap fixed the loading for packages.

**Comment by Xerxes-2** [comment]
I think newly installed homebrew doesn't have the default taps now, so the tap list is empty.

**Comment by buresdv** [comment]
> I think newly installed homebrew doesn't have the default taps now, so the tap list is empty.

That's whack on Homebrew's part, I will test it in a VM

**Comment by buresdv** [comment]
You were right, it doesn't have any taps by default. I'll add a way of optionally adding them for the user, see #151 for progress on that.

What's weird though is that the packages from `homebrew/core` and `homebrew/cask` can still be found and installed, even when those aren't tapped.

**Comment by buresdv** [comment]
Fixed as per #151
